### DISCUSSION TITLE
Deployed bytecode is not equal to the generated one from a given source

### DISCUSSION THREAD

**Discussion by lbeder** [zkSync-Community-Hub/zksync-developers#507] [Contract Verification]
### Environment

Mainnet

### Issue Type

- [X] Plugin
- [x] Block Explorer

### Contract Address

0x21d8d4eE83b80bc0Cc0f2B7df3117Cf212d02901

### Compiler Type

Multi-part contract

### zkSolc Version

1.4.1

### Solc Version

0.8.19

### Contract Name

EAS

### Contract Code

Please here: https://github.com/ethereum-attestation-service/eas-contracts/blob/zksync/contracts/EAS.sol

### Constructor Arguments

Please here: https://github.com/ethereum-attestation-service/eas-contracts/blob/zksync/contracts/EAS.sol

### Hardhat Verify Plugin Version

1.4.2

### Repo Link (Optional)

https://github.com/ethereum-attestation-service/eas-contracts/tree/zksync

### Additional Details

We deployed 4 EAS contracts, but the main EAS contracts cannot be verified. They deployed and complied, but only one contract cannot be verified (our main contract). We also tried verifying in the UI, but that didn't work either. Seems to be some syntax mismatch?

ErrorCode:
ZkSyncVerifyPluginError: Backend verification error: Deployed bytecode is not equal to the generated one from a given source

You can find the verification script here: https://github.com/ethereum-attestation-service/eas-contracts/blob/zksync/scripts/verify.ts, but please note that manual verification using the UX (using a flattened version) didn't work either. 

Related branch of our deployment:
https://github.com/ethereum-attestation-service/eas-contracts/tree/zksync

**Comment by bxpana** [comment]
Any luck with verifying them via the Block Explorer? https://explorer.zksync.io/contracts/verify

Are any of the contracts you're trying to verify, deployed by another contract? There is a known issue with not being able to verify contracts deployed by other contracts right now.

**Reply by lbeder** [reply]
> Any luck with verifying them via the Block Explorer? https://explorer.zksync.io/contracts/verify

Nope. It's exactly the same result when trying to verify via the Block Explorer. 3 contracts work well, and only one fails with the same error.

> Are any of the contracts you're trying to verify, deployed by another contract? There is a known issue with not being able to verify contracts deployed by other contracts right now.

All the contracts are deployed using an EOA and exactly in the same way.

**Reply by bxpana** [reply]
Alright, will check on my end to see if I can replicate.

**Reply by lbeder** [reply]
Thanks 🙏

**Comment by nikola-bozin-txfusion** [comment]
Hey @lbeder , we've verified your contract: https://explorer.zksync.io/address/0x21d8d4eE83b80bc0Cc0f2B7df3117Cf212d02901#contract. 
:smile: 
We'll address the issue on the plugin side.

**Reply by lbeder** [reply]
Hey @nikola-bozin-txfusion, thanks for verifying, but have you noticed that you basically include all the contracts in the repo and not only the contracts used by EAS.sol? Could you please update it to only include the relevant contracts? Otherwise, it looks kind of looks weird and unprofessional :/

**Reply by nikola-bozin-txfusion** [reply]
When resolving contracts for verification of EAS Smart Contract, hardhat gathers dependent contracts. 
It seems that is not enough for zksolc compiler (compilation output is different - bytecode) on BE for the verification, hence it fallsback to sending the the full source.
We will be working with the backend/compiler team to resolve this issue asap.

**Reply by lbeder** [reply]
Yeah, I completely understand why you have fallen back to an equivalent of etherscan''s `--solc-input`, but it currently makes the main contract appear bad. Please keep up posted when it's resolved so that we could reverify.

**Reply by lbeder** [reply]
Hey @nikola-bozin-txfusion, any news on this? Is it possible to properly re-verify the contracts?

**Comment by hedgar2017** [comment]
Already reported in the `solc` repo, but I've sent a link to this thread just in case:
https://github.com/ethereum/solidity/issues/15134

**Comment by hedgar2017** [comment]
@lbeder according to the [solc v0.8.26 release notes](https://soliditylang.org/blog/2024/05/21/solidity-0.8.26-release-announcement/), a bug like that was fixed in 0.8.26, so I recommend trying that version when we release a `zksolc` with its support.

**Reply by lbeder** [reply]
Hi @hedgar2017, this isn't relevant to our issue.
